src/daemon.rs

```diff
@@ -253,3 +253,54 @@ pub(crate) fn extract_bitcoind_error(err: &bitcoincore_rpc::Error) -> Option<&Rp
         _ => None,
     }
 }
+
+// Bitcoin RPC error codes
+// https://github.com/bitcoin/bitcoin/blob/master/src/rpc/protocol.h
+#[derive(Debug, PartialEq, FromPrimitive, ToPrimitive)]
+pub(crate) enum RPCErrorCode {
+    RpcInvalidRequest = -32600,
+    RpcMethodNotFound = -32601,
+    RpcInvalidParams = -32602,
+    RpcInternalError = -32603,
+    RpcParseError = -32700,
+
+    RpcMiscError = -1,
+    RpcTypeError = -3,
+    RpcInvalidAddressOrKey = -5,
+    RpcOutOfMemory = -7,
+    RpcInvalidParameter = -8,
+
+    RpcDatabaseError = -20,
+    RpcDeserializationError = -22,
+    RpcVerifyError = -25,
+    RpcVerifyRejected = -26,
+    RpcVerifyAlreadyInChain = -27,
+    RpcInWarmup = -28,
+    RpcMethodDeprecated = -32,
+
+    RpcClientNotConnected = -9,
+    RpcClientInInitialDownload = -10,
+    RpcClientNodeAlreadyAdded = -23,
+    RpcClientNodeNotAdded = -24,
+    RpcClientNodeNotConnected = -29,
+    RpcClientInvalidIpOrSubnet = -30,
+    RpcClientP2pDisabled = -31,
+    RpcClientNodeCapacityReached = -34,
+    RpcClientMempoolDisabled = -33,
+
+    RpcWalletError = -4,
+    RpcWalletInsufficientFunds = -6,
+    RpcWalletInvalidLabelName = -11,
+    RpcWalletKeypoolRanOut = -12,
+    RpcWalletUnlockNeeded = -13,
+    RpcWalletPassphraseIncorrect = -14,
+    RpcWalletWrongEncState = -15,
+    RpcWalletEncryptionFailed = -16,
+    RpcWalletAlreadyUnlocked = -17,
+    RpcWalletNotFound = -18,
+    RpcWalletNotSpecified = -19,
+    RpcWalletAlreadyLoaded = -35,
+    RpcWalletAlreadyExisits = -36,
+
+    RpcForbiddenBySafeMode = -2,
```

Comment:
can be removed as this was reserved in a version lower than the minimum supported bitcoind version of electrs